lib/internal/repl/history.js

```diff
@@ -141,7 +141,7 @@ function setupHistory(repl, historyPath, ready) {
       return;
     }
     writing = true;
-    const historyData = ArrayPrototypeJoin(repl.history, os.EOL);
+    const historyData = ArrayPrototypeJoin(repl.history, '\n');
```

Comment:
How would the history be handled in case the file already contains `\r\n`? I fear existing history files on windows would break.

Reply:
I just tested it, this is how it behaves:

if you write history with the old node and you open it with the new node, this is how it looks like:

![image](https://github.com/user-attachments/assets/f01fefb4-beac-4bc4-99e5-43f7612e4e3c)

aka, it sees a new additional line at the end of each line.

if you go back tho to the old node, the history is displayed correctly, without new lines.

To be honest, I don't see it too bad

Reply:
I believe we could optimize this away. As in: if it's just a single extra line, it should just be skipped. I believe a user can not really create such input at all.

Reply:
@puskin94 did you have a look at this one again? :)
I tried to look at it in the code but I am uncertain if it's still the case or not and I currently have no Windows for testing.

Reply:
Yessir, it is fixed! I had to change the regex from `/\n+/` to `/\r?\n+/`